lib/package/sns/compileMethodsToSns.js

```diff
@@ -0,0 +1,147 @@
+          template,
+          this.getSnsMethodIntegration(event.http),
+          this.getMethodResponses(event.http)
+        )
+
+        const methodLogicalId = this.provider.naming.getMethodLogicalId(
+          resourceName,
+          event.http.method
+        )
+
+        this.apiGatewayMethodLogicalIds.push(methodLogicalId)
+
+        _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, {
+          [methodLogicalId]: template
+        })
+      }
+    })
+  },
+
+  getSnsMethodIntegration(http) {
+    const integration = {
+      IntegrationHttpMethod: 'POST',
+      Type: 'AWS',
+      Credentials: {
+        'Fn::GetAtt': ['ApigatewayToSnsRole', 'Arn']
+      },
+      Uri: {
+        'Fn::Join': [
+          '',
+          [
+            'arn:aws:apigateway:',
+            {
+              Ref: 'AWS::Region'
+            },
+            ':sns:path//'
+          ]
+        ]
+      },
+      PassthroughBehavior: 'NEVER',
+      RequestParameters: {
+        'integration.request.header.Content-Type': "'application/x-www-form-urlencoded'"
+      },
+      RequestTemplates: this.getSnsIntegrationRequestTemplates(http)
+    }
+
+    const integrationResponse = {
+      IntegrationResponses: [
+        {
+          StatusCode: 200,
+          SelectionPattern: 200,
+          ResponseParameters: {},
+          ResponseTemplates: {}
+        },
+        {
+          StatusCode: 400,
+          SelectionPattern: 400,
+          ResponseParameters: {},
+          ResponseTemplates: {}
+        },
+        {
+          StatusCode: 500,
+          SelectionPattern: 500,
+          ResponseParameters: {},
+          ResponseTemplates: {}
+        }
+      ]
+    }
+
+    if (http && http.cors) {
+      let origin = http.cors.origin
+      if (http.cors.origins && http.cors.origins.length) {
+        origin = http.cors.origins.join(',')
+      }
+
+      integrationResponse.IntegrationResponses.forEach(async (val, i) => {
+        integrationResponse.IntegrationResponses[i].ResponseParameters = {
+          'method.response.header.Access-Control-Allow-Origin': `'${origin}'`
+        }
+      })
+    }
+
+    _.merge(integration, integrationResponse)
+
+    return {
+      Properties: {
+        Integration: integration
+      }
+    }
+  },
+
+  getSnsIntegrationRequestTemplates(http) {
+    const defaultRequestTemplates = this.getDefaultSnsRequestTemplates(http)
+    return Object.assign(defaultRequestTemplates, _.get(http, ['request', 'template']))
+  },
+
+  getDefaultSnsRequestTemplates(http) {
+    return {
+      'application/json': this.buildDefaultSnsRequestTemplate(http),
+      'application/x-www-form-urlencoded': this.buildDefaultSnsRequestTemplate(http)
+    }
+  },
+
+  buildDefaultSnsRequestTemplate(http) {
+    const { topicName } = http
+
+    const topicArn = {
+      'Fn::Sub': ['arn:aws:sns:${AWS::Region}:${AWS::AccountId}:${topicName}', { topicName }]
+    }
+
+    return {
+      'Fn::Join': [
```

Comment:
same here, can we use `Fn::Sub` instead?

Reply:
Sure